### PR TITLE
improve frame management and resolve mouse position into frames

### PR DESCRIPTION
**Pull request by jefleponot**
Its is impossible to use mouse.click, mouse.move method when there are frames into pages.

This PR goals : 
- Solved mouse event on frames
- Improve Frame management with new methods
- Increases tests for Frames

### CODE REVIEW

**Comment by jefleponot** [comment]
@hlr1983

Could you review my code, please ... ?

File: modules/casper.js [hunk 1]
```diff
@@ -2204,6 +2213,105 @@ function getTimeoutAndCheckNextStepFunction(timeout, then, methodName, defaultTi
...
+ * Makes the provided frame page as the currently active one. Note that the
+ * active page will NOT be reverted when finished.
+ *
+ * @param  String|Number frameInfo  Target frame name or number
+ * @return Casper
+ */
+Casper.prototype.switchToFrame = function switchToFrame(frameInfo) {
+    "use strict";
+    this.switchToChildFrame(frameInfo);
+    this.page.frameLevel++;
```
Comment (hlr1983) [inline modules/casper.js]: Should add a `this.page.frameLevel--;` to `switchToParentFrame()` ?

File: modules/casper.js [hunk 2]
```diff
@@ -1078,7 +1077,7 @@ Casper.prototype.getElementsAttr = function getElementsAttr(selector, attribute)
  * @param  String  selector  A DOM CSS3/XPath selector
  * @return Object
```
Comment (hlr1983) [inline modules/casper.js]: Add a `@param page`
Also change the docs.

File: modules/casper.js (hunk 1 above)
Comment (hlr1983) [inline modules/casper.js]: Should add a `this.page.frameLevel--;` to `switchToParentFrame()` ?
And `this.page.frameLevel = 0;` to `switchToMainFrame()` ?

**Comment by jefleponot** [comment]
@hlr1983 

Thanks a lot :smile:

**Pull request merged**